bip-musig2/reference.py

```diff
@@ -333,7 +335,7 @@ def sign(secnonce: bytes, sk: bytes, session_ctx: SessionContext) -> bytes:
     assert R_2_ is not None
     pubnonce = cbytes(R_1_) + cbytes(R_2_)
     # Optional correctness check. The result of signing should pass signature verification.
-    assert partial_sig_verify_internal(psig, pubnonce, bytes_from_point(P), session_ctx)
+    assert partial_sig_verify_internal(psig, pubnonce, cbytes(P), session_ctx)
     return psig
 
 def partial_sig_verify(psig: bytes, pubnonces: List[bytes], pubkeys: List[bytes], tweaks: List[bytes], is_xonly: List[bool], msg: bytes, i: int) -> bool:
```

Comment:
Should `pubkeys` be renamed to `plain_pks` or something?

Reply:
I changed the type to `List[PlainPk]` and while doing that realized that mypy did not detect the resulting type errors in the randomized tests. Apparently mypy only checks functions that are typed. So I added the types which led mypy to detect a few type errors even in the parent branch. A bunch of them were fixed by adding a `keygen` function.